File: BitwiseXorOfAllPairings/Leetcode2425.java
public class Leetcode2425 {
    public int xorAllNums(int[] nums1, int[] nums2) {
        if(nums1.length == 0 && nums2.length == 0) return 0;

        int x = nums1.length;
        int y = nums2.length;

        if(x % 2 == 0 && y % 2 == 0) {
            return 0;
        } else if(x % 2 == 0) {
            return getXor(nums1);
        } else if(y % 2 == 0) {
            return getXor(nums2);
        } else {
            int xor1 = getXor(nums1);
            int xor2 = getXor(nums2);
            return xor1 ^ xor2;
        }
    }

    private int getXor(int[] arr) {
        int result = arr[0];
        for(int i = 1; i < arr.length; i++) {
            result ^= arr[i];
        }
        return result;
    }
}

File: BitwiseXorOfAllPairings/Leetcode2425.cpp
#include<vector>
using std::vector;

class Leetcode2425 {
    int getXor(const vector<int>& v) {
        int result = v.at(0);
        for(auto it = v.begin() + 1; it < v.end(); ++it) {
            result ^= *it;
        }
        return result;
    }
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        size_t x = nums1.size();
        size_t y = nums2.size();
        if(x % 2 == 0 && y % 2 == 0) return 0;
        else if(x % 2 == 0) return getXor(nums1);
        else if(y % 2 == 0) return getXor(nums2);
        else return getXor(nums1) ^ getXor(nums2);
    }
};